app/services/homepage/fetch_articles.rb

```diff
@@ -0,0 +1,32 @@
+module Homepage
+  class FetchArticles
+    DEFAULT_PER_PAGE = 60
+
+    def self.call(
+      approved: nil, published_at: nil, sort_by: nil, sort_direction: nil, page: 0, per_page: DEFAULT_PER_PAGE
+    )
+      articles = Homepage::ArticlesQuery.call(
+        approved: approved,
+        published_at: published_at,
+        sort_by: sort_by,
+        sort_direction: sort_direction,
+        page: page,
+        per_page: per_page,
+      )
+
+      # Unfortunately the FlareTag class sends one SQL query per each article,
+      # as we want to optimize by loading them in one query, we're using a different class
+      tag_flares = Homepage::FetchTagFlares.call(articles)
```

Comment:
hacking the night away. Initially I had N queries based on how many articles there were, now only 2 for this statement, indipendently from how many articles are passed in